packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -89,3 +113,77 @@ function makeWrappedErrorLogger(origErrorLogger: ErrorLogger): WrappedErrorLogge
     return origErrorLogger.call(this, err);
   };
 }
+
+/**
+ * Wrap the server's request handler to be able to create request transactions.
+ *
+ * @param origReqHandler The original request handler from the `Server` class
+ * @returns A wrapped version of that handler
+ */
+function makeWrappedReqHandler(origReqHandler: ReqHandler): WrappedReqHandler {
+  const liveServer = closure.server as Server;
+
+  // inspired by
+  // https://github.com/vercel/next.js/blob/4443d6f3d36b107e833376c2720c1e206eee720d/packages/next/next-server/server/next-server.ts#L1166
+  const publicDirFiles = new Set(
+    deepReadDirSync(liveServer.publicDir).map(p =>
+      encodeURI(
+        // switch any backslashes in the path to regular slashes
+        p.replace(/\\/g, '/'),
+      ),
+    ),
+  );
+
+  // add transaction start and stop to the normal request handling
+  const wrappedReqHandler = async function(
+    this: Server,
+    req: NextRequest,
+    res: NextResponse,
+    parsedUrl?: url.UrlWithParsedQuery,
+  ): Promise<void> {
+    // We only want to record page and API requests
+    if (hasTracingEnabled() && shouldTraceRequest(req.url, publicDirFiles)) {
+      const transaction = Sentry.startTransaction({
+        name: `${(req.method || 'GET').toUpperCase()} ${req.url}`,
+        op: 'http.server',
+      });
+      Sentry.getCurrentHub()
+        .getScope()
+        ?.setSpan(transaction);
+
+      res.__sentry__ = {};
+      res.__sentry__.transaction = transaction;
```

Comment:
```suggestion
      res.__sentry__ = { transaction };
```

Reply:
or if you want to preserve previous value (which shouldn't even be there In the first place anyway):

```suggestion
      res.__sentry__ = res.__sentry__ || {};
      res.__sentry__.transaction = transaction;
```

Reply:
Yeah, definitely. It's only like this because in the next PR I'm going to initialize it with actual stuff in it.